### PR TITLE
[Integration][Datadog] Add a new kind of SLO History

### PR DESCRIPTION
**Pull request by phalbert**
# Description

**What**
Get SLO [History](https://docs.datadoghq.com/api/latest/service-level-objectives/#get-an-slos-history)

**Why**

**How**

## Type of change

Please leave one option from the following and delete the rest:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] New Integration (non-breaking change which adds a new integration)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Non-breaking change (fix of existing functionality that will not change current behavior)
- [ ] Documentation (added/updated documentation)

## Screenshots

Include screenshots from your environment showing how the resources of the integration will look.

## API Documentation

Provide links to the API documentation used for this integration.

### CODE REVIEW

File: integrations/datadog/.port/resources/blueprints.json [hunk 1]
```diff
@@ -194,5 +202,39 @@
...
+    "mirrorProperties": {},
+    "calculationProperties": {},
+    "relations": {
+      "monitors": {
+        "target": "datadogSlo",
+        "title": "SLO",
+        "description": "The SLO to which this history belongs to",
+        "required": false,
+        "many": true
+      }
```
Comment (Tankilevitch) [inline integrations/datadog/.port/resources/blueprints.json:237]: looks from the mapping that its one to one relation and should be required I guess

File: integrations/datadog/.port/resources/port-app-config.yaml [hunk 2]
```diff
@@ -57,3 +57,21 @@ resources:
           relations:
             monitors: .monitor_ids | map(tostring)
             services: '.monitor_tags + .tags | map(select(startswith("service:"))) | unique | map(split(":")[1])'
+  - kind: sloHistory
+    selector:
+      query: 'true'
+      fromTimestamp: 1711411200 # The from timestamp for the query window in epoch seconds.
+      toTimestamp: 1711898238
```
Comment (Tankilevitch) [inline integrations/datadog/.port/resources/port-app-config.yaml:64]: how is this expected to be used in the mapping structure? will this need to be updated every day? I think we need a more useable solution? please elaborate

**Review by Tankilevitch — CHANGES_REQUESTED** [review]

File: integrations/datadog/client.py [hunk 3]
```diff
@@ -172,6 +173,26 @@ async def get_slos(self) -> AsyncGenerator[list[dict[str, Any]], None]:
...
 
+    async def list_slo_histories(
+        self, from_ts: int, to_ts: int
+    ) -> AsyncGenerator[list[dict[str, Any]], None]:
+        histories = []
+        async for slos in self.get_slos():
+            histories = await asyncio.gather(
+                *(self.get_slo_history(slo["id"], from_ts, to_ts) for slo in slos)
+            )
+            yield histories
```
Comment (yairsimantov20) [inline integrations/datadog/client.py:184]: will it help to stream this requests?

Comment (yairsimantov20) [reply integrations/datadog/client.py:184]: can you fetch multiple get_slo_history together in the same request?

File: integrations/datadog/CHANGELOG.md [hunk 4]
```diff
@@ -7,6 +7,18 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 <!-- towncrier release notes start -->
 
+# Port_Ocean 0.1.12 (2024-06-30)
```
Comment (yairsimantov20) [inline integrations/datadog/CHANGELOG.md:10]: update

File: integrations/datadog/client.py [hunk 5]
```diff
@@ -172,6 +173,26 @@ async def get_slos(self) -> AsyncGenerator[list[dict[str, Any]], None]:
...
+
+    async def get_slo_history(
+        self, slo_id: str, from_ts: int, to_ts: int
+    ) -> dict[str, Any]:
+        url = f"{self.api_url}/api/v1/slo/{slo_id}/history"
+        result = await self._send_api_request(
+            url, params={"from_ts": from_ts, "to_ts": to_ts}
+        )
+        history = result.get("data")
+        return history
```
Comment (yairsimantov20) [inline integrations/datadog/client.py:194]:
```suggestion
        return result["data"]
```

**Review by yairsimantov20 — APPROVED** [review]
🌊

File: integrations/datadog/CHANGELOG.md (hunk 4 above)
Comment (Tankilevitch) [inline integrations/datadog/CHANGELOG.md:10]:
```suggestion
# Port_Ocean 0.1.12 (2024-07-01)
```

File: integrations/datadog/client.py (hunk 3 above)
Comment (Tankilevitch) [reply integrations/datadog/client.py:184]: stream won't help, we want to return in batches

**Review by Tankilevitch — APPROVED** [review]

**Pull request merged**